### PR TITLE
Installation docs for aarch64-apple-darwin

### PR DESCRIPTION
**Pull request by Hoverbear**
Similar to #664, but Apple flavored instead of glass-flavored.

### CODE REVIEW

File: ferrocene/doc/user-manual/src/targets/aarch64-apple-darwin.rst [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+.. warning::
+   
+   Experimental targets cannot be used in safety-critical contexts, and there is
+   no guarantee that the Ferrocene test suite is successfully executed on the
+   target. They are provided as a preview, with limited support available. They
+   should not be used in production.
+
+The ``aarch64-apple-darwin`` Ferrocene target provides support for macOS (Darwin) on
+aarch64.
```
Comment (pietroalbini) [inline ferrocene/doc/user-manual/src/targets/aarch64-apple-darwin.rst:17]: Nit: I'd use Apple Silicon here.

File: ferrocene/doc/user-manual/src/targets/aarch64-apple-darwin.rst [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+-----------------------
+
+To use the target, the following additional flags must be provided to
+``rustc``:
+
+- ``--target=x86_64-unknown-linux-gnu``
+
+- ``-C linker=<your-c-compiler>``
+
+  - e.g. ``-C linker=/usr/bin/clang``
```
Comment (pietroalbini) [inline ferrocene/doc/user-manual/src/targets/aarch64-apple-darwin.rst:71]: Do we actually need users to specify this?

Comment (Hoverbear) [reply ferrocene/doc/user-manual/src/targets/aarch64-apple-darwin.rst:71]: Don't think so, but the Linux docs specify it so I did here.

Comment (pietroalbini) [reply ferrocene/doc/user-manual/src/targets/aarch64-apple-darwin.rst:71]: Response is the same as the Windows comment.

**Review by pietroalbini — APPROVED** [review]
bors merge

**Comment by Hoverbear** [comment]
I'll fix this up.

**Pull request merged**